src/flows/pipes/Markdown/MarkdownPanel.tsx

```diff
@@ -63,9 +64,7 @@ const MarkdownPanel: FC<PipeProp> = ({Context}) => {
     <MarkdownModeToggle mode={data.mode} onToggleMode={handleToggleMode} />
   )
 
-  const handleChange = (text: string): void => {
-    update({text})
-  }
+  const handleChange = debounce((text: string): void => update({text}), 500)
```

Comment:
why does this help? shouldn't we have a clear functional path from the input event, through the local storage event cycle, and back up through the rendering pipeline to display the value through props?

I'm worried that this is brushing any thing to the contrary under the rug, and is only going to make it harder to debug when these updates are being funneled through the network interface from a bunch of users

Reply:
Debounce saves from unnecessary re-renders! 

Without debounce:

https://user-images.githubusercontent.com/6411855/109856944-f105af00-7c0e-11eb-93ff-14894a047948.mov

With debounce:

https://user-images.githubusercontent.com/6411855/109856928-ec40fb00-7c0e-11eb-90de-7a9bfd44d318.mov

Reply:
I will look further into why we have this weird behavior with multiple tabs without debounce, but I think regardless debounce is a worthy performance improvement that we should use

Reply:
lemme know what you find. still leaning more towards that "hides rendering issues" than "fixes rendering issues", but totally open to new info

Reply:
for those reading this in a not-so-distant-future: the issue was with `use-persisted-state`. Switching libraries fixed the issue and removed the need for debounce.